src/components/GoSingleFileInput/index.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useCallback } from 'react';
+import { _cs } from '@togglecorp/fujs';
+import { NameType, ValueType, OptionKey } from '#components/types';
+import { CloseLineIcon } from '@ifrc-go/icons';
+import IconButton, { Props as IconButtonProps } from '#components/IconButton';
+import Button, { Props as ButtonProps } from '#components/Button';
+import { ButtonVariant } from '#hooks/useButtonFeatures';
+import RawFileInput, { RawFileInputProps } from '#components/RawFileInput';
+import { useLazyRequest } from '#utils/restRequest';
+import { nonFieldError } from '@togglecorp/toggle-form';
+import useAlert from '#hooks/useAlert';
+
+import styles from './styles.module.css';
+
+export type Props<T extends NameType, O, V extends ValueType, K extends OptionKey> = Omit<RawFileInputProps<T>, 'multiple' | 'value' | 'onChange' | 'children'> & {
+    actions?: React.ReactNode;
+    buttonProps?: ButtonProps<T>;
+    children?: React.ReactNode;
+    className?: string;
+    clearButtonProps?: IconButtonProps<T>;
+    clearable?: boolean;
+    icons?: React.ReactNode;
+    iconsClassName?: string;
+    keySelector: (option: O) => K;
+    onChange: (value: K | undefined | null, name: T) => void;
+    setFileIdToUrlMap?: React.Dispatch<React.SetStateAction<Record<K, V>>>;
+    url: string;
```

Comment:
Similar comment to the GoMultiFileInput about the url, keySelector and valueSElector